### PR TITLE
feat: Unified gateway deployment image. Closes #639

### LINKED ISSUE #639: Unified gateway image
**Is your feature request related to a problem? Please describe.**
Different type of gateway has its own image, this is super hard to manage.

**Describe the solution you'd like**
Wonder if we can have a unified image with different args to start gateway deployment.

### PR DESCRIPTION
**Pull request by whynowy**
Unified image for all the gateways, start the service with an argument to specify the type.

Closes #639

This commit also updated the Dockerfiles for gateway-controller, sensor-controller and gateway-client to use `scratch`, this reduces the image size by 70-80% and brings more secure.

### CODE REVIEW

File: controllers/sensor/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,7 @@
-FROM centos:7
+FROM alpine:latest as certs
+RUN apk --update add ca-certificates
+
+FROM scratch
```
Comment (whynowy) [inline controllers/sensor/Dockerfile:4]: Please note the Dockerfile change.

File: Makefile [hunk 2]
```diff
@@ -94,259 +94,17 @@ gateway-client-image:
...
-	go build -v -ldflags '${LDFLAGS}' -o ${DIST_DIR}/azure-events-hub-gateway ./gateways/server/azure-events-hub/cmd
-
-azure-events-hub-linux:
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 make azure-events-hub
-
-azure-events-hub-image:
-	@if [ "$(BUILD_BINARY)" = "true" ]; then make azure-events-hub-linux; fi
-	docker build -t $(IMAGE_PREFIX)azure-events-hub-gateway:$(IMAGE_TAG) -f ./gateways/server/azure-events-hub/Dockerfile .
-	@if [ "$(DOCKER_PUSH)" = "true" ] ; then  docker push $(IMAGE_PREFIX)azure-events-hub-gateway:$(IMAGE_TAG) ; fi
+gateway-server-image:
```
Comment (VaibhavPage) [inline Makefile:104]: love the slimmed down Makefile.

File: controllers/sensor/Dockerfile (hunk 1 above)
Comment (VaibhavPage) [reply controllers/sensor/Dockerfile:4]: 👍

Comment (VaibhavPage) [reply controllers/sensor/Dockerfile:4]: We need git.

**Comment by whynowy** [comment]
@VaibhavPage - Thanks! Before I click the `merge` button, could you please drop a `stable` tag? we need to update at least the installation yaml file link, otherwise this will PR will break it.

**Comment by VaibhavPage** [comment]
Can we make the yaml file changes part of another PR just before next release? So you can merge the code changes.

**Pull request merged**